lib/src/write_task.dart

```diff
@@ -9,8 +9,8 @@ enum WriteCommand {
 }
 
 class WriteTask {
-  WriteCommand command;
-  DocumentReference document;
-  Map<String, dynamic> data;
-  bool merge;
+  WriteCommand? command;
```

Comment:
Is it possible to have null here?
(I don't feel so)

Reply:
You are right. I thought setting it as non-nullable would force me to define a constructor, but I can use the `late` keyword instead. It's much cleaner now, thank you! Making it non-nullable as it should be, allowed me to remove lots of unnecessary `!`:

![image](https://user-images.githubusercontent.com/3399854/111739571-9e8ee880-88c6-11eb-96f2-fcd31d8648fe.png)